esmf_regrid/tests/unit/schemes/_common_regridder.py

```diff
@@ -0,0 +1,283 @@
+
+    def test_differing_grids(self):
+        """Test erroring when called on a different grid to the init grid."""
+        n_lons = 6
+        n_lats = 5
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        src_init = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+        tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+        n_lons_dif = 7
+        src_dif_coord = _grid_cube(
+            n_lons_dif, n_lats, lon_bounds, lat_bounds, circular=True
+        )
+        src_dif_circ = _grid_cube(
+            n_lons, n_lats, lon_bounds, lat_bounds, circular=False
+        )
+
+        regridder = self.REGRIDDER(src_init, tgt)
+
+        msg = "The given cube is not defined on the same source grid as this regridder."
+        with pytest.raises(ValueError, match=msg):
+            _ = regridder(src_dif_coord)
+        with pytest.raises(ValueError, match=msg):
+            _ = regridder(src_dif_circ)
+
+    def test_invalid_mdtol(self):
+        """Test erroring when mdtol is out of range."""
+        n_lons = 6
+        n_lats = 5
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        src = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+        tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+        match = "Value for mdtol must be in range 0 - 1, got "
+        with pytest.raises(ValueError, match=match):
+            _ = self.REGRIDDER(src, tgt, mdtol=2)
+        with pytest.raises(ValueError, match=match):
+            _ = self.REGRIDDER(src, tgt, mdtol=-1)
+
+    def test_curvilinear_equivalence(self):
+        """Test curvilinear and rectilinear coordinates giving the same results."""
+        n_lons_src = self.N_LONS_SRC
+        n_lons_tgt = 3
+        n_lats_src = 4
+        n_lats_tgt = 2
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        grid_src = _grid_cube(
+            n_lons_src, n_lats_src, lon_bounds, lat_bounds, circular=True
+        )
+        grid_tgt = _grid_cube(
+            n_lons_tgt, n_lats_tgt, lon_bounds, lat_bounds, circular=True
+        )
+        curv_src = _curvilinear_cube(n_lons_src, n_lats_src, lon_bounds, lat_bounds)
+        curv_tgt = _curvilinear_cube(n_lons_tgt, n_lats_tgt, lon_bounds, lat_bounds)
+
+        grid_to_grid = self.REGRIDDER(grid_src, grid_tgt)
+        grid_to_curv = self.REGRIDDER(grid_src, curv_tgt)
+        curv_to_grid = self.REGRIDDER(curv_src, grid_tgt)
+        curv_to_curv = self.REGRIDDER(curv_src, curv_tgt)
+
+        def extract_weights(regridder):
+            return regridder.regridder.weight_matrix.todense()
+
+        for regridder in [grid_to_curv, curv_to_grid, curv_to_curv]:
+            assert np.allclose(
+                extract_weights(grid_to_grid), extract_weights(regridder)
+            )
+
+    def test_curvilinear_and_rectilinear(self):
```

Comment:
Note that this test diverges in behaviour with #276 due to differing behavour of regridders with respect to bounds.